widgetssdk/src/main/java/com/glia/widgets/helper/ActivityWatcherForDialogs.kt

```diff
@@ -71,42 +79,202 @@ class ActivityWatcherForDialogs(
         dialogController?.removeCallback(dialogCallback)
     }
 
+    private fun addScreenSharingCallback(resumedActivity: Activity?) {
+        setupScreenSharingViewCallback(resumedActivity)
+        screenSharingController?.setViewCallback(screenSharingViewCallback)
+        screenSharingController?.onResume(resumedActivity)
+    }
+
+    private fun removeScreenSharingCallback() {
+        screenSharingController?.removeViewCallback(screenSharingViewCallback)
+    }
+
+    private fun setupScreenSharingViewCallback(resumedActivity: Activity?) {
+        screenSharingViewCallback = resumedActivity?.let {
+            ScreenSharingController.ViewCallback { exception: GliaException ->
+                showToast(it, exception.debugMessage)
+            }
+        }
+    }
+
     @VisibleForTesting
     fun setupDialogCallback(resumedActivity: Activity?) {
         dialogCallback = DialogController.Callback {
             when (it.mode) {
                 Dialog.MODE_NONE -> dismissAlertDialog()
+                Dialog.MODE_OVERLAY_PERMISSION -> resumedActivity?.runOnUiThread {
+                    showOverlayPermissionsDialog(resumedActivity)
+                }
                 Dialog.MODE_MEDIA_UPGRADE -> resumedActivity?.runOnUiThread {
                     showUpgradeDialog(resumedActivity, it as DialogState.MediaUpgrade)
                 }
+                Dialog.MODE_START_SCREEN_SHARING -> resumedActivity?.runOnUiThread {
+                    showScreenSharingDialog(resumedActivity)
+                }
+                Dialog.MODE_ENABLE_NOTIFICATION_CHANNEL -> resumedActivity?.runOnUiThread {
+                    showAllowNotificationsDialog(resumedActivity)
+                }
+                Dialog.MODE_ENABLE_SCREEN_SHARING_NOTIFICATIONS_AND_START_SHARING -> resumedActivity?.runOnUiThread {
+                    showAllowScreenSharingNotificationsAndStartSharingDialog(resumedActivity)
+                }
             }
         }
     }
 
+    private fun showOverlayPermissionsDialog(resumedActivity: Activity) {
+        if (alertDialog != null && alertDialog!!.isShowing) {
```

Comment:
as an example of the previous comment
`val activity = resumedActivity?.get() ?: return`
and then use `activity` instance